sanic/request.py

```diff
@@ -179,20 +179,34 @@ async def respond(
                 headers=headers,
                 content_type=content_type,
             )
+
+        # Check the status of current stream and response.
+        re_use_res = False
+        try:
+            if self.stream is not None:
+                if self.stream.stage is Stage.IDLE:
+                    raise ResponseException(
+                        "Another response was sent previously."
+                    )
+                re_use_res = response is (self.stream and self.stream.response)
+        except AttributeError:
+            pass
+
         # Connect the response
         if isinstance(response, BaseHTTPResponse) and self.stream:
             response = self.stream.respond(response)
         # Run response middleware
-        try:
-            response = await self.app._run_response_middleware(
-                self, response, request_name=self.name
-            )
-        except CancelledErrors:
-            raise
-        except Exception:
-            error_logger.exception(
-                "Exception occurred in one of response middleware handlers"
-            )
+        if not re_use_res:
```

Comment:
I like this better. Let's just use a better variable name: `should_reuse_response`